packages/tracking/src/events/types.ts

```diff
@@ -224,15 +224,8 @@ export type UserVisitData = UserSharedData & {
   is_ugc?: boolean;
 };
 
-export type UserImpressionData = {
-  page_name: string;
```

Comment:
should we keep `page_name` and `target` here since they are required? they both exist in `UserSharedData` but they're optional there

Reply:
Using a union with `UserSharedData` also adds optional properties we probably don't want on this event (at least at the moment since the [BE event](https://github.com/codecademy-engineering/Codecademy/blob/f3ec2b611789dbdf21115b2228696befc7166dd4/lib/segment_tracker/user_impression_event.rb) won't add them): `href`, `version`, `source_codebase`. Perhaps using [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) would be a better option?

Reply:
@aresnik11 good catch!